src/composables/state.ts

```diff
@@ -0,0 +1,497 @@
+import VueCompositionAPI, { computed, ref } from '@vue/composition-api'
+import RoutingSlipService from '@/services/routingSlip.services'
+import { AccountInfo, AdjustRoutingSlipAmountPrams, AdjustRoutingSlipChequePrams, GetRoutingSlipRequestPayload, LinkedRoutingSlips, RoutingSlip, RoutingSlipDetails } from '@/models/RoutingSlip'
+import { ApiErrors, CreateRoutingSlipStatus, SlipStatus } from '@/util/constants'
+import CodesService from '@/services/codes.service'
+import CommonUtils from '@/util/common-util'
+import { BusinessInfo, GetFeeRequestParams, Payment, TransactionParams } from '@/models/Payment'
+import Vue from 'vue'
+import { Code } from '@/models/Code'
+
+Vue.use(VueCompositionAPI)
+
+// Routing Slips
+// Refs
+export const headerSearchTitle = ref([
+  {
+    text: 'Routing Slip Number',
+    align: 'start',
+    value: 'routingSlipNumber',
+    display: true,
+    className: 'routing-slip'
+  },
+  {
+    text: 'Receipt Number',
+    align: 'start',
+    sortable: false,
+    value: 'receiptNumber',
+    display: true,
+    className: 'receiptNumber'
+  },
+  {
+    text: 'Entity Number',
+    align: 'start',
+    value: 'accountName',
+    sortable: false,
+    display: false,
+    className: 'accountName'
+  },
+  {
+    text: 'Created By',
+    align: 'start',
+    value: 'createdName',
+    sortable: false,
+    display: false,
+    className: 'createdName'
+  },
+  {
+    text: 'Date',
+    align: 'start',
+    sortable: false,
+    value: 'date',
+    display: true,
+    className: 'date'
+  },
+  {
+    text: 'Status',
+    align: 'start',
+    sortable: false,
+    value: 'status',
+    display: true,
+    className: 'status'
+  },
+  {
+    text: 'Reference Number',
+    align: 'start',
+    value: 'businessIdentifier',
+    sortable: false,
+    display: true,
+    className: 'businessIdentifier'
+  },
+  {
+    text: 'Cheque Number',
+    align: 'start',
+    value: 'chequeReceiptNumber',
+    sortable: false,
+    display: false,
+    className: 'cheque-receipt-number'
+  },
+  {
+    text: 'Balance',
+    align: 'right',
+    value: 'remainingAmount',
+    sortable: false,
+    display: true,
+    className: 'remainingAmount'
+  },
+  {
+    text: 'Actions',
+    align: 'start',
+    value: '',
+    sortable: false,
+    display: true,
+    hideInSearchColumnFilter: true,
+    className: 'action'
+  }
+])
+export const searchRoutingSlipResult = ref<RoutingSlip[]>([])
+export const searchRoutingSlipParams = ref<any>({})
+
+export const routingSlip = ref<RoutingSlip>({})
+export const linkedRoutingSlips = ref<LinkedRoutingSlips>(undefined)
+export const routingSlipDetails = ref<RoutingSlipDetails>({})
+export const accountInfo = ref<AccountInfo>({})
+export const chequePayment = ref<Payment[]>([])
+export const cashPayment = ref<Payment>({})
+export const isPaymentMethodCheque = ref(true)
+export const isAmountPaidInUsd = ref(false)
+export const autoCompleteRoutingSlips = ref<RoutingSlip[]>([])
```

Comment:
it would be nice to have the 3 'stores' in separate files and returned as composables (i.e. useRoutingSlip, useCodes, useIndicators). That way you can put variables/computed's in reactive objects which allow you to reference things without '.value' within the composable and its easier to come in and change things when you have the 'state' of each one as a typed object (especially when multiple people are updating it)

Reply:
I'll separate "state" into three stores. 

I didn't use reactive objects.. wanted to keep it similar to what was there before (hence the refs) - I'd like to leave this for another PR if that's possible

Reply:
yeah definitely 👍